cpp/src/arrow/io/interfaces.h

```diff
@@ -191,7 +191,7 @@ class ARROW_EXPORT ReadWriteFileInterface : public RandomAccessFile, public Writ
 };
 
 // TODO(kszucs): remove this after 0.13
```

Comment:
It seems like the right thing to do might be to delete the using statements altogher based on the TODO above? @kszucs?

Reply:
Agreed. I was unsure what "this" referred to in the comment. The whole ifndef and else, or part of it. How long after 0.13 is required? What is it about post-0.13 that makes it possible to remove, etc.

Reply:
Agreed with @emkornfield, we should simply remove those aliases.

Reply:
Agreed

Reply:
Thanks everyone. Can you explain what from this code block you would like me to remove, specifically? I'm new here. I'm just trying to eliminate potentially build-breaking warnings for RAPIDS libcudf.

https://github.com/apache/arrow/blob/1a3eab50ea49414494d4f7ed91d1e2e3a6f8919f/cpp/src/arrow/io/interfaces.h#L193-L201

Reply:
These codes are for users who use old Arrow C++. We can remove codes for backward compatibility when we release two or more minor (or major) versions. So we can just apply the following diff:

```diff
diff --git a/cpp/src/arrow/io/interfaces.h b/cpp/src/arrow/io/interfaces.h
index 3a5cfe3d7..678366bd7 100644
--- a/cpp/src/arrow/io/interfaces.h
+++ b/cpp/src/arrow/io/interfaces.h
@@ -190,16 +190,6 @@ class ARROW_EXPORT ReadWriteFileInterface : public RandomAccessFile, public Writ
   ReadWriteFileInterface() { RandomAccessFile::set_mode(FileMode::READWRITE); }
 };
 
-// TODO(kszucs): remove this after 0.13
-#ifndef _MSC_VER
-using WriteableFile ARROW_DEPRECATED("Use WritableFile") = WritableFile;
-using ReadableFileInterface ARROW_DEPRECATED("Use RandomAccessFile") = RandomAccessFile;
-#else
-// MSVC does not like using ARROW_DEPRECATED with using declarations
-using WriteableFile = WritableFile;
-using ReadableFileInterface = RandomAccessFile;
-#endif
-
 }  // namespace io
 }  // namespace arrow
```

Reply:
Thanks @kou. Done.